api/api/views/media_views.py

```diff
@@ -267,6 +267,10 @@ def related(self, request, identifier=None, *_, **__):
 
         serializer_context = self.get_serializer_context()
 
+        serializer_class = self.get_serializer()
+        if serializer_class.needs_db:
+            results = self.get_db_results(results)
```

Comment:
Do we have any idea of how much getting the db results affects the performance of the searches?
I wonder how we can measure if it's better to get this data from the database, or to keep it in ES indexes and get it from there?

Reply:
It is a single query to the DB for all results using the field `identifier` that's  both `unique=True` and `index=True`. I don't think it is inefficient. Personally I prefer one the way it is rn because we have to query the DB anyway for relational info that can't be stored in ES. Also the amount of relational info will only increase with new projects slated for 2024.

Reply:
Thank you for the explanation, @dhruvkb , makes sense!

Reply:
@dhruvkb 100% in agreement here as well, that would be a good thing to document somewhere as the "rationale" of the serialiser approach, prefereably in code or, if not there, then in the API docs, in something of a "design decisions" document. It's similar to the the ES document `_id` not being the same as `identifier`, for example, in that it's easy to forget why this is the "right" way to do it and then re-ask this same question (or for new folks to ask this question). A "design decisions" doc I think would be a great thing to have in general (to summarise my point :grinning:).

Reply:
Actually, rather than a new documentation page, pulling this behaviour into a function that the two view methods can call and adding the rationale in the doc string there would be more resilient to future changes.

Reply:
Both views ultimately use the `get_db_results` function, I've documented that function in 324fe92. I think the `needs_db` mechanism is largely moot because almost all operations ultimately need the DB and once that is simplified, `get_db_results` will be the common functionality entirely.